exporter/signalfxexporter/config.go

```diff
@@ -160,7 +156,7 @@ func (cfg *Config) validateConfig() error {
 		return errors.New(`cannot have a negative "timeout"`)
 	}
 
-	if cfg.MaxConnections < 0 {
+	if cfg.MaxConnections != nil && *cfg.MaxConnections < 0 {
```

Comment:
Would probably be a good idea to log a warning that this field is deprecated if not nil and they should use the client settings fields.

Reply:
Pushed the changes accordingly.